x/erc20/keeper/genesis.go

```diff
@@ -0,0 +1,121 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/evmos/evmos/v16/x/erc20/types"
+	"github.com/evmos/evmos/v16/x/evm/statedb"
+)
+
+// SetGenesisTokenPairs stores in state the provided token pairs in genesis
+func (k Keeper) SetGenesisTokenPairs(ctx sdk.Context, pairs []types.TokenPair) error {
+	for i, pair := range pairs {
+		contractAddr := pair.GetERC20Contract()
+		coinMeta, err := k.getGenesisTokenPairMeta(ctx, pair, i)
+		if err != nil {
+			return fmt.Errorf("error while generating metadata for genesis pair denom %s. %w", pair.Denom, err)
+		}
+		stateDB := statedb.New(ctx, k.evmKeeper, statedb.TxConfig{})
```

Comment:
second thought. This is a very low level call. Let's only use the EVM keeper functions (`Call`, `SetCode`, etc) instead